lucene/core/src/test/org/apache/lucene/search/TestIndexSearcher.java

```diff
@@ -70,13 +75,31 @@ public void setUp() throws Exception {
     }
     reader = iw.getReader();
     iw.close();
+
+    Random random = random();
+    RandomIndexWriter iw2 = new RandomIndexWriter(random(), dir2, newIndexWriterConfig().setMergePolicy(newLogMergePolicy()));
+    for (int i = 0; i < 100; i++) {
+      Document doc = new Document();
+      doc.add(newStringField("field", Integer.toString(i), Field.Store.NO));
+      doc.add(newStringField("field2", Boolean.toString(i % 2 == 0), Field.Store.NO));
+      doc.add(new SortedDocValuesField("field2", new BytesRef(Boolean.toString(i % 2 == 0))));
+      iw2.addDocument(doc);
+
+      if (random.nextBoolean()) {
+        iw2.commit();
+      }
+    }
+    reader2 = iw2.getReader();
+    iw2.close();
```

Comment:
why are we not reusing reader1?

Reply:
reader2 has random commits baked into it to ensure random slice distributions each time the test is run.

Reply:
Would it be a problem to add these random commits to the existing reader?

Reply:
I did not want to touch the existing reader since it is used by many tests -- dont see any risk though, given the specifics of the tests that use the existing reader, should be safe. Updated, thanks